src/array/cpu/spmm_blocking_libxsmm.h

```diff
@@ -0,0 +1,577 @@
+/*!
+ *  Copyright (c) 2021 Intel Corporation
+ * \file array/cpu/spmm.h
+ * \brief SPMM CPU kernel function header.
+ * \author Sanchit Misra <sanchit.misra@intel.com>,
+ *         Ramanarayan Mohanty <ramanarayan.mohanty@intel.com>,
+ *         Vasimuddin Md <vasimuddin.md@intel.com>,
+ *         Sasikanth Avancha <sasikanth.avancha@intel.com>
+ */
+#ifndef DGL_ARRAY_CPU_SPMM_BLOCKING_LIBXSMM_H_
+#define DGL_ARRAY_CPU_SPMM_BLOCKING_LIBXSMM_H_
+
+#include <dgl/array.h>
+#include <dgl/bcast.h>
+#include <dmlc/logging.h>
+#include <algorithm>
+
+#if !defined(_WIN32)
+#ifdef USE_AVX
+#ifdef USE_LIBXSMM
+#include <unistd.h>
+#include <libxsmm.h>
+#ifdef DEBUG
+#include <x86intrin.h>
+#endif
+#include <omp.h>
+
+namespace dgl {
+namespace aten {
+namespace cpu {
+
+template <typename IdType, typename DType>
+  struct CSRM{
+    IdType num_rows;
+    IdType num_cols;
+    IdType *indptr;
+    IdType *indices;
+    DType *data;
+  };
+
+int32_t get_llc_size() {
+  int32_t cache_size = sysconf(_SC_LEVEL3_CACHE_SIZE);
+  if(cache_size < 0) cache_size = DGL_CPU_LLC_SIZE;
+  return cache_size;
+}
+
+
+template <typename IdType, typename DType>
+inline void SpMMCreateBlocks(
+    const CSRMatrix& csr,
+    CSRM<IdType, IdType> *block_csr_array,
+    IdType num_M_blocks,
+    IdType num_K_blocks,
+    IdType M_BLOCK_SIZE,
+    IdType K_BLOCK_SIZE,
+    DType *C, IdType N,
+    bool use_lhs, bool use_rhs) {
+
+  const IdType M = csr.num_rows;
+  const IdType K = csr.num_cols;
+  IdType* IndPtr = csr.indptr.Ptr<IdType>();
+  IdType* Indices = csr.indices.Ptr<IdType>();
+  IdType* Edges = csr.data.Ptr<IdType>();
+  assert(IndPtr != nullptr);
+  if(use_lhs)
+    assert(Indices != nullptr);
+  if(use_rhs)
+    assert(Edges != nullptr);
+
+  if(num_K_blocks > 1)
+  {
+    IdType *indptr = (IdType *)aligned_alloc(64, (M_BLOCK_SIZE + 1) * num_M_blocks * num_K_blocks * sizeof(IdType));
+
+#pragma omp parallel
+    {
+      IdType *my_cur_col_id = (IdType *)aligned_alloc(64, 2 * M_BLOCK_SIZE * sizeof(IdType));
+
+#pragma omp for
+      for(IdType m = 0; m < num_M_blocks; m++)
```

Comment:
Add a space between `for` and parenthesis:
```suggestion
      for (IdType m = 0; m < num_M_blocks; m++)
```

Reply:
Done